base/reducedim.jl

```diff
@@ -258,8 +258,9 @@ function _mapreducedim!(f, op, R::AbstractArray, A::AbstractArrayOrBroadcasted)
         # use mapreduce_impl, which is probably better tuned to achieve higher performance
         nslices = div(length(A), lsiz)
         ibase = first(LinearIndices(A))-1
-        for i = 1:nslices
-            @inbounds R[i] = op(R[i], mapreduce_impl(f, op, A, ibase+1, ibase+lsiz))
+        for i = range(firstindex(R), length=nslices)
```

Comment:
I think this would be even better:

```suggestion
        for i in eachindex(R)
```

Reply:
That's certainly better. I hadn't looked into whether the range spans the entire axis

Reply:
Yeah, I wouldn't have known this had I not been neck deep in reductions lately. That's the biggest advantage here IMV — using eachindex brings so much more clarity to what is happening here, which really isn't that complicated.